### DISCUSSION TITLE
Clerk and Adonis authentication code example

### DISCUSSION THREAD

**Discussion by ivanvinicius** [clerk/javascript#3393] [Support request - DO NOT OPEN HERE] [answered]
I'm trying to find some example of authentication with Clerk and Adonis. But I didn't find any.

**Does anyone know a repository, git-gist or forum thread that can help me?**

Any example will be useful.

I'm using adonis v5.8 and the latest version of clerk.

thanks!

**Comment by wobsoriano** [comment, marked as answer]
Check this example - https://github.com/wobsoriano/adonis-clerk-example

There is a custom Clerk [Middleware](https://github.com/wobsoriano/adonis-clerk-example/blob/main/app/middleware/clerk_middleware.ts) that'll inject an auth object to the http context:

```ts
router.get('/api/me', (ctx) => {
  const { userId } = ctx.request.auth

  if (!userId) {
    return ctx.response.status(401).send('Unauthorized')
  }

  return ctx.response.json(ctx.request.auth)
})
```

**Reply by wobsoriano** [reply]
Also, please feel free to improve it since I'm not an Adonis expert :D

**Reply by ivanvinicius** [reply]
Thank you :D
